### PR TITLE
ALL tests passing in spec harness for item and item repo

### PR DESCRIPTION
**Pull request by ochar721**

### CODE REVIEW

**Comment by kylejschulz** [comment]
Hey, let us know how it looks. There will be merge conflicts with merchant and merchant repo in main but we don't need to keep any of our stuff.

File: lib/item_repository.rb [hunk 1]
```diff
@@ -65,12 +71,12 @@ def find_all_by_merchant_id(merchant_id)
 
   def create(attributes)
     id = @items[-1].id.to_i
+
     id += 1
     id = id.to_s
     attributes[:id] = id
```
Comment (Pragmaticpraxis37) [inline lib/item_repository.rb:77]: Maybe a helper method here would be appropriate?

Comment (jequeza) [reply lib/item_repository.rb:77]: Agree, we can have a refactoring session soon!

File: lib/item_repository.rb [hunk 2]
```diff
@@ -1,30 +1,41 @@
...
+      @items << Item.new(row, self)
     end
   end
 
   def all
-    @items.length
+    @items
   end
 
   def find_by_id(id)
```
Comment (Pragmaticpraxis37) [inline lib/item_repository.rb:36]: Maybe we can factor this out?

File: lib/item_repository.rb [hunk 3]
```diff
@@ -1,30 +1,41 @@
...
     @items.find do |item|
```
Comment (Pragmaticpraxis37) [inline lib/item_repository.rb:37]: Should we factor this one out?

**Review by Pragmaticpraxis37 — APPROVED** [review]
Looks great all!

**Pull request merged**